src/lib/server/bitcoin-nodeless.ts

```diff
@@ -32,27 +32,30 @@ export function isZPubValid(zpub: string): boolean {
 }
 
 export async function generateDerivationIndex(): Promise<number> {
-	const res = await collections.runtimeConfig.findOneAndUpdate(
-		{ _id: 'bitcoinNodeless' },
-		{
-			$inc: { 'data.derivationIndex': 1 as never },
-			$set: { updatedAt: new Date() }
-		},
-		{ returnDocument: 'before' }
-	);
-
-	if (!res.value) {
-		throw new Error('Failed to increment derivation index');
+	let index = runtimeConfig.bitcoinNodeless.derivationIndex;
+	
+	for (let attempts = 0; attempts < 10; attempts++) {
+		const address = bip84Address(runtimeConfig.bitcoinNodeless.publicKey, index);
+		const isUsed = await isAddressUsed(address).catch(() => false);
+		
+		if (!isUsed) {
+			await collections.runtimeConfig.updateOne(
+				{ _id: 'bitcoinNodeless' },
+				{ $set: { 'data.derivationIndex': index + 1, updatedAt: new Date() } }
+			);
```

Comment:
The pattern of updating both the database and inmemory object of `runtimeConfig` is pretty common.

We should at least use `persistConfigElement`.

In a separate commit we could improve `persistConfigElement` to return the persisted value **and** update `runtimeConfig`.